ons-spark/raw-notebooks/date-functions/date-functions.ipynb

```diff
@@ -642,89 +569,89 @@
-       "      <td>2010-12-24</td>\n",
+       "      <td>2014-12-24</td>\n",
        "      <td>Christmas Eve</td>\n",
        "      <td>3</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>6</th>\n",
-       "      <td>2010-12-25</td>\n",
-       "      <td>Christmas Day</td>\n",
+       "      <td>2016-12-24</td>\n",
+       "      <td>Christmas Eve</td>\n",
        "      <td>1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>7</th>\n",
-       "      <td>2010-12-26</td>\n",
-       "      <td>Boxing Day</td>\n",
-       "      <td>3</td>\n",
+       "      <td>2018-12-24</td>\n",
+       "      <td>Christmas Eve</td>\n",
+       "      <td>2</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>8</th>\n",
-       "      <td>2011-01-01</td>\n",
-       "      <td>New Years Day</td>\n",
-       "      <td>2</td>\n",
+       "      <td>2009-12-25</td>\n",
+       "      <td>Christmas Day</td>\n",
+       "      <td>1</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>9</th>\n",
-       "      <td>2011-12-24</td>\n",
-       "      <td>Christmas Eve</td>\n",
-       "      <td>2</td>\n",
+       "      <td>2010-12-25</td>\n",
+       "      <td>Christmas Day</td>\n",
+       "      <td>1</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "text/plain": [
        "  date_of_call    festive_day  count\n",
-       "0   2009-01-01  New Years Day      2\n",
-       "1   2009-12-24  Christmas Eve      1\n",
-       "2   2009-12-25  Christmas Day      1\n",
-       "3   2009-12-26     Boxing Day      2\n",
-       "4   2010-01-01  New Years Day      2\n",
-       "5   2010-12-24  Christmas Eve      3\n",
-       "6   2010-12-25  Christmas Day      1\n",
-       "7   2010-12-26     Boxing Day      3\n",
-       "8   2011-01-01  New Years Day      2\n",
-       "9   2011-12-24  Christmas Eve      2"
+       "0   2009-12-24  Christmas Eve      1\n",
+       "1   2010-12-24  Christmas Eve      3\n",
+       "2   2011-12-24  Christmas Eve      2\n",
+       "3   2012-12-24  Christmas Eve      1\n",
+       "4   2013-12-24  Christmas Eve      1\n",
+       "5   2014-12-24  Christmas Eve      3\n",
+       "6   2016-12-24  Christmas Eve      1\n",
+       "7   2018-12-24  Christmas Eve      2\n",
+       "8   2009-12-25  Christmas Day      1\n",
+       "9   2010-12-25  Christmas Day      1"
       ]
      },
-     "execution_count": 8,
+     "execution_count": 20,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "festive_rescues.groupBy(\"date_of_call\", \"festive_day\").count().orderBy(\"date_of_call\").limit(10).toPandas()"
+    "festive_rescues.groupBy(\"date_of_call\", \"call_month_day\").count().orderBy(\"call_month_day\",\"date_of_call\").replace(festive_dates).withColumnRenamed(\"call_month_day\",\"festive_day\").limit(10).toPandas()"
```

Comment:
This is a lot to have on a single line, can we enclose it in brackets and go to separate lines? Something like this:

(festive_rescues.groupBy("date_of_call", "call_month_day")
 .count()
 .orderBy("call_month_day","date_of_call")
 .replace(festive_dates)
 .withColumnRenamed("call_month_day","festive_day")
 .limit(10).toPandas())